src/clib/lib/enkf/queue_config.cpp

```diff
@@ -55,7 +55,7 @@ static queue_config_type *queue_config_alloc_empty() {
     queue_config->driver_type = NULL_DRIVER;
     queue_config->user_mode = false;
     queue_config->max_submit = 2; // Default value
-    queue_config->num_cpu = 0;
```

Comment:
I think this change would be the correct if this code was following normal conventions. However, I think something non-standard is going on. num_cpu=0 is clearly not correct, but perhaps this is signalling some sort of "don't force the number of cpus" behavior. I think we will have to dig a bit to see what is going on before we make a change here.

Reply:
i'm gonna try to investigate a bit then! thanks for pointing it out.

Reply:
found the first hint in `src/ert/ensemble_evaluator/builder/_ensemble_builder.py`, in the function `from_legacy`, lines 124 ff:

```
    num_cpu = res_config.queue_config.num_cpu
    if num_cpu == 0:
        num_cpu = res_config.ecl_config.num_cpu
```

Reply:
and we have the same pattern in `src/ert/_c_wrappers/job_queue/queue.py`, in `add_job_from_run_arg`:

```
    num_cpu = res_config.queue_config.num_cpu
    if num_cpu == 0:
        num_cpu = res_config.ecl_config.num_cpu
```

what do you think about the solution I proposed above, @eivindjahren ?

Reply:
that seem to be all instances of usages of `queue_config`'s `num_cpu`

Reply:
It seems like this code is signaling that 0 means use `res_config.ecl_config.num_cpu`. I think we should clean that up in a separate PR. Isn't it possible to just recreate how this number is instantiated in this PR and then clean this particular behavior up separately?

Reply:
i'll see if i can get away in this PR without those changes. i think something made me introduce them, but perhaps it was just confusion ; )

Reply:
I have removed that commit, and rather pushed it with changes as discussed here in #3919

Reply:
oh and i fixed the test that was failing and triggered me to do the changes in question to work without those changes - basically just needed to make sure that the same value for `NUM_CPU` was present both in config file and config dict (the former was not the case)